app/src/main/java/com/nutomic/syncthingandroid/activities/WebGuiActivity.java

```diff
@@ -56,6 +64,19 @@
      */
     private final WebViewClient mWebViewClient = new WebViewClient() {
 
+        /**
+         * Catch errors like the timeout on loading a web page.
+         */
+         public void onReceivedError(WebView view, int errorCode, String description, String failingUrl) {
+             if (errorCode == WebViewClient.ERROR_TIMEOUT) {
+                Log.w(TAG, "Timeout loading locally running web UI. Retrying ...");
+                if (setWebViewProxy(mWebView.getContext().getApplicationContext(), "", 0, "localhost|0.0.0.0|127.*|[::1]")) {
+                    view.reload();
```

Comment:
Put a return after the reload and verified on my phone. It works without calling the base class' error handler after view.reload. Thanks for pointing it out.

Reply:
Doesn't this cause an endless loop in case of proxy failure?
If the connection times out, we set the proxy, which succeeds, we reload which then causes a timeout again, etc.

Reply:
Thanks for the review. I will test an alternative tomorrow and report back if the endless loop can be improved.

Reply:
Isn't it enough just to set it on the web view upon construction?

Reply:
Unfortunately no, tested this before and found it not working by a chance of ~80%. Thats why I put the reload and setproxy on timeout a second time. Maybe there is some delay in the app to webview "communication" or delay on the webview handling the updated proxy settings. Im sure we find a solution on that...

Reply:
80% chance on what. This is not quantum computing, our code is deterministic.